spring-grpc-core/src/main/java/org/springframework/grpc/client/DefaultGrpcChannelFactory.java

```diff
@@ -70,6 +70,18 @@ public DefaultGrpcChannelFactory(List<GrpcChannelBuilderCustomizer<T>> globalCus
 		this.interceptorsConfigurer = interceptorsConfigurer;
 	}
 
+	/**
+	 * Whether this factory supports the given target string. The target can be either a
+	 * valid nameresolver-compliant URI, an authority string as described in
+	 * {@link Grpc#newChannelBuilder(String, ChannelCredentials)}.
+	 * @param target the target string as described in method javadocs
+	 * @return true unless the target begins with 'in-process:'
+	 */
+	@Override
+	public boolean supports(String target) {
+		return !target.startsWith("in-process:");
```

Comment:
I wonder if we should use "directaddress" instead of "in-process"? It's what gRPC calls it natively.

Reply:
Unfortunately, the `InProcessChannelBuilder` will not strip that prefix off if its already there. So we would end up still having to do the stripping and then the name `directaddress` is further from the name of the feature. Because of this, I vote to keep it as-is. Wdyt?